docs/source/install.rst

```diff
@@ -294,40 +274,167 @@ should run::
+* ``complex``: an installation where PETSc is configured using complex numbers
+
+The different configurations can be selected by passing the flag ``--arch`` to
+``firedrake-configure``. For example::
+
+   $ python3 firedrake-configure --show-petsc-configure-options --arch complex
+
+
+Optional dependencies
+---------------------
+
+PETSc with 64-bit indices
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To build PETSc with 64-bit indices you need to append ``--with-64-bit-indices``
+to the set of options when PETSc is configured. For example this could be::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options | xargs -L1 ./configure --with-64-bit-indices
+
+SLEPc
+~~~~~
+
+To install Firedrake with SLEPc support you must:
+
+#. Clone and install SLEPc (note that ``PETSC_DIR`` and ``PETSC_ARCH`` must be set)::
+
+   $ git clone --depth 1 https://github.com/firedrakeproject/slepc.git
+   $ cd slepc
+   $ ./configure
+   $ make SLEPC_DIR=$PWD
+
+#. Set ``SLEPC_DIR``::
+
+   $ export SLEPC_DIR=$PWD
+
+#. Install Firedrake with SLEPc support::
+
+   $ pip install --no-binary h5py "firedrake @ git+https://github.com/firedrakeproject/firedrake.git#[test,slepc]"
+
+VTK
+~~~
+
+To install Firedrake with VTK, it should be installed using the ``vtk`` optional
+dependency. For example::
+
+   $ pip install --no-binary h5py "firedrake @ git+https://github.com/firedrakeproject/firedrake.git#[test,vtk]"
 
-   $ make reconfigure
+
+PyTorch
+~~~~~~~
+
+To install Firedrake with `PyTorch <https://pytorch.org/>`_, it should be installed
+using the ``torch`` optional dependency. For example::
+
+   $ pip install --no-binary h5py "firedrake @ git+https://github.com/firedrakeproject/firedrake.git#[test,torch]" --extra-index-url https://download.pytorch.org/whl/cpu
+
+Observe that, in addition to specifying ``torch``, an additional
+argument (``--extra-index-url``) is needed. More information on installing
+PyTorch can be found `here <https://pytorch.org/get-started/locally/>`__.
+
+
+Netgen
+~~~~~~
+
+To install Firedrake with `Netgen <https://ngsolve.org/>`_ support, it should be
+installed with the ``netgen`` optional dependency. For example::
+
+   $ pip install --no-binary h5py "firedrake @ git+https://github.com/firedrakeproject/firedrake.git#[test,netgen]"
+
+
+Advanced customisation
+----------------------
+
+Since ``firedrake-configure`` only outputs a string of options it is straightforward
+to customise all aspects of the installation process. You can either:
+
+* Append additional options when ``configure`` is invoked. For example::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options | xargs -L1 ./configure --download-exotic-package
+
+* Write the output of ``firedrake-configure`` to a file than can be modified. For example::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options > my_configure_options.txt
+   <edit my_configure_options.txt>
+   $ ./configure $(cat my_configure_options.txt)
 
```

Comment:
```suggestion

For better reproducibility of PETSc builds, PETSc allows you to write a python script to record the configuration options and call configure. Example scripts can be found in the `config/examples/` directory of the PETSc repository.
```

Reply:
I don't know if this is an essential detail. It's just another way to achieve the same thing as
```
$ ./configure $(cat myoptions.txt)
```
And I don't want to overload the user with information

Reply:
How about just reducing it to something smaller like:
"You can see examples of different PETSc configurations in the `petsc/config/examples` directory. Be aware that these will not necessarily have all requirements for Firedrake"
I do think it's good to direct to the config python scripts if people are modifying the configuration themselves.

Reply:
I've added something.